deluser, userdel: add page (#5245)
tldr-pages/tldr: pages/linux/deluser.md

# deluser

> > Delete user from system

Reviewer: ```suggestion
> Delete a user from them system.
```

> > Delete user from system.

Reviewer: Perhaps this will read better?
```suggestion
> Delete a user from the system.
```

> > Note: all commands must be executed as root.

Reviewer: ```suggestion
> Note: all commands must be executed as root.
> More information: <https://manpages.debian.org/deluser.8.en.html>.
```

Author: Sure, we can add this. Not all people have debian tho, and I thought that man pages are kinda too obvious

Reviewer: Well, we do that if there's no other page.

Reviewer: If there's something better than the Debian manpage go ahead and use it.  We use man7.org when possible, then I guess the Debian manpages.

> > All commands must be executed as root (`sudo [command]`)

Reviewer: ```suggestion
> Note: all commands must be executed as root.
```

> > All commands must be executed as root (`sudo [command]`).

Reviewer: ```suggestion
> Note: all commands must be executed as root.
```

- Remove a user:

> `deluser {{name}}`

Reviewer: A user might be confused and think display name, e.g. so they might run `deluser John Doe` instead of `deluser johndoe` ;)
```suggestion
`deluser {{username}}`
```

- Remove a user and their home directory:

> `deluser --remove-home {{name}}`

Reviewer: ```suggestion
`deluser --remove-home {{username}}`
```

> - Remove a user and their home, but backup it's files first:

Reviewer: ```suggestion
- Remove a user and their home, backing up their files:
```

Author: I didn't know how to put this well, but I would keep keep it as original (if it's not strongly against the rules)

Reviewer: "it's" doesn't sound right&mdash;if you're referring to the user, I would use "their".

Reviewer: Oh, also https://github.com/tldr-pages/tldr/pull/5245#discussion_r571437570

Reviewer: How about this?
```suggestion
- Remove a user and their home, but backup their files into a `.tar.gz` file in the specified directory:
```

Author: I tried playing with it, and I discovered that pure `--backup` doesn't work and `--backup-to` needs to have exisitng directory (can't create a new one)

Should we add this?

Reviewer: I'm not sure what you mean&mdash;my suggestion seems good enough.  If the directory is required to exist, well, I don't think we need to state that...many commands require the specified file/directory to exist (obviously :smile:)

> `deluser --backup-to {{path/to/backup}} --remove-home {{name}}`

Reviewer: I assume this is a directory?
```suggestion
`deluser --backup-to {{path/to/backup_directory}} --remove-home {{username}}`
```

Author: Oh, yes, it's a directory where one `.tag.gz` file will be placed

Reviewer: Oh it's an archive?  Then I would clarify this in the description.

- Remove a user, and all files owned by them:

> `deluser --remove-all-files {{name}}

Reviewer: This is a cool option!

> `deluser --remove-all-files {{name}}`

Reviewer: ```suggestion
`deluser --remove-all-files {{username}}`
```

This is a cool option!! :smile:

# deluser

> Delete a user from the system.
> Note: all commands must be executed as root.

> > More information: <https://manpages.debian.org/deluser.8.en.html>.

Reviewer 2: This link is not working for me.

```suggestion
> More information: <https://manpages.debian.org/adduser/deluser.8.html>.
```

Reviewer: Strange. In all browsers it redirects to https://manpages.debian.org/buster/adduser/deluser.8.en.html

But adding `adduser/` is a bit more verbose&mdash;for the better, IMO.

- Remove a user:

`deluser {{username}}`

- Remove a user and their home directory:

`deluser --remove-home {{username}}`

> 

Reviewer 2: Could you add an extra example for `--backup`:
```suggestion

- Remove a user and their home, but backup their files into a `.tar.gz` file in the current directory:

`deluser --backup --remove-home {{username}}`

```

Reviewer: Actually @navarroaxel, it goes to `/` not the cwd.

Reviewer 2: My mistake. Check the description now.

Reviewer: How about just `in /`?

Author: > I tried playing with it, and I discovered that pure `--backup` doesn't work [...]

I got no absolutley no files in / when I was using it :shrug: Ubuntu 20.04
